spyder_line_profiler/spyder/plugin.py

```diff
@@ -0,0 +1,133 @@
+# Copyright © 2021, Spyder Line Profiler contributors
+#
+# Licensed under the terms of the MIT license
+# ----------------------------------------------------------------------------
+"""
+Spyder Line Profiler 5 Plugin.
+"""
+
+# Third-party imports
+from qtpy.QtGui import QIcon
+
+# Spyder imports
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.api.plugin_registration.decorators import on_plugin_available
+from spyder.api.plugin_registration.decorators import on_plugin_teardown
+from qtpy.QtCore import Signal
+
+# Local imports
+from spyder.plugins.mainmenu.api import ApplicationMenus
+from spyder_line_profiler.spyder.confpage import SpyderLineProfilerConfigPage
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidget
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidgetActions
+from spyder_line_profiler.spyder.widgets import is_lineprofiler_installed
+
+_ = get_translation("spyder_line_profiler.spyder")
+
+
+
+class SpyderLineProfilerActions:
+    # Triggers
+    Run = 'run_action_menu'
+    
+
+class SpyderLineProfiler(SpyderDockablePlugin):
+    """
+    Spyder Line Profiler plugin for Spyder 5.
+    """
+
+    NAME = "spyder_line_profiler"
+    REQUIRES = [Plugins.Editor, Plugins.Help]
+    OPTIONAL = [Plugins.MainMenu]
+    TABIFY = [Plugins.Help]
+    WIDGET_CLASS = SpyderLineProfilerWidget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfilerConfigPage
+
+    # --- Signals
+    sig_finished = Signal()
+    """This signal is emitted to inform the profile profiling has finished."""
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    @staticmethod
+    def get_name():
+        return _("Line Profiler")
+
+    def get_description(self):
+        return _("Line profiler display for Spyder 5")
+
+    def get_icon(self):
+        return QIcon('./data/images/spyder.line_profiler.png')
+
+    def on_initialize(self):
+        self.widget = self.get_widget()
+        self.widget.sig_finished.connect(self.sig_finished)
+        
+        run_action = self.create_action(
+                        SpyderLineProfilerActions.Run,
+                        text=_("Run line profiler"),
+                        tip=_("Run line profiler"),
+                        icon=self.get_icon(),
+                        triggered=self.run_lineprofiler,
+                        register_shortcut=True,
+                        )   
+        run_action.setEnabled(is_lineprofiler_installed())
+        
+        
+    @on_plugin_available(plugin=Plugins.MainMenu)
+    def on_main_menu_available(self):
+        mainmenu = self.get_plugin(Plugins.MainMenu)
+        start_action = self.get_action(SpyderLineProfilerActions.Run)
+        mainmenu.add_item_to_application_menu(
+            start_action, menu_id=ApplicationMenus.Run)
```

Comment:
```suggestion
        run_action = self.get_action(SpyderLineProfilerActions.Run)
        mainmenu.add_item_to_application_menu(
            run_action, menu_id=ApplicationMenus.Run)
```

Use a more telling name for this variable.